app/views/unattended/provisioning_templates/provision/kickstart_default.erb

```diff
@@ -111,85 +111,22 @@ selinux --<%= host_param('selinux-mode') || host_param('selinux') || 'enforcing'
 keyboard <%= host_param('keyboard') || 'us' %>
 
 <%
-  network_options = []
-  nameservers = []
-  subnet4 = iface.subnet
-  subnet6 = iface.subnet6
-
-  # device and hostname
-  if iface.bond? && rhel_compatible && os_major >= 6
-    network_options.push("--device=#{iface.identifier}")
-  else
-    network_options.push("--device=#{iface.mac || iface.identifier}")
-  end
-  network_options.push("--hostname #{@host.name}")
-
-  # single stack
-  if subnet4 && !subnet6
-    network_options.push("--noipv6")
-  elsif !subnet4 && subnet6
-    network_options.push("--noipv4")
-  end
-
-  # dual stack MTU check
-  raise("IPv4 and IPv6 subnets have different MTU") if subnet4 && subnet6 && subnet4.mtu.present? && subnet6.mtu.present? && subnet4.mtu != subnet6.mtu
-
-  # IPv4
-  if (subnet4 && !subnet4.dhcp_boot_mode?) || @static
-    network_options.push("--bootproto static")
-    network_options.push("--ip=#{iface.ip}")
-    network_options.push("--netmask=#{subnet4.mask}")
-    network_options.push("--gateway=#{subnet4.gateway}")
-  elsif subnet4 && subnet4.dhcp_boot_mode?
-    network_options.push("--bootproto dhcp")
-  end
-  if subnet4
-    nameservers.concat(subnet4.dns_servers)
-    network_options.push("--mtu=#{subnet4.mtu}") if subnet4.mtu.present?
-  end
-
-  # IPv6
-  if rhel_compatible && os_major >= 6
-    if (subnet6 && !subnet6.dhcp_boot_mode?) || @static6
-      network_options.push("--ipv6=#{iface.ip6}/#{subnet6.cidr}")
-      network_options.push("--ipv6gateway=#{subnet6.gateway}")
-    elsif subnet6 && subnet6.dhcp_boot_mode?
-      if host_param_true?('use-slaac')
-        network_options.push("--ipv6 auto")
-      else
-        network_options.push("--ipv6 dhcp")
-      end
-    end
-    if subnet6
-      nameservers.concat(subnet6.dns_servers)
-      network_options.push("--mtu=#{subnet6.mtu}") if subnet6.mtu.present?
-    end
-  end
-
-  # bond
-  if iface.bond? && rhel_compatible && os_major >= 6
-    bond_slaves = iface.attached_devices_identifiers.join(',')
-    network_options.push("--bondslaves=#{bond_slaves}")
-    network_options.push("--bondopts=mode=#{iface.mode},#{iface.bond_options.tr(' ', ',')}")
-  end
-
-  # VLAN (only on physical is recognized)
-  if iface.virtual? && iface.tag.present? && iface.attached_to.present?
-    if rhel_compatible && os_major == 6
-      network_options.push("--vlanid=#{iface.tag}")
-    else
-      network_options.push("--interfacename=vlan#{iface.tag}")
-    end
-  end
-
-  # DNS
-  if nameservers.size > 0
-    network_options.push("--nameserver=#{nameservers.join(',')}")
-  else
-    network_options.push("--nodns")
-  end
+# start with provisioning interface, then other non-bond interfaces and the bonds at the end
+@host.managed_interfaces.sort_by { |iface| iface.bond? ? 0 : iface.provision? ? 20 : 10 }.each do |iface|
+-%>
+<%= snippet(
```

Comment:
I have a host (RHEL 9) with the following interfaces:
```
Network interfaces: 
 1) Id:           12
    Identifier:   leo1
    Type:         interface (primary, provision)
    MAC address:  00:aa:bb:cc:72:b6
    IPv4 address: 192.66.66.3
    IPv6 address: 2574:bbd0:91c0:79f9:d421:3bb9:401d:3212
    FQDN:         floyd-obanion.example.com
 2) Id:           13
    Identifier:   leo2
    Type:         interface (virtual attached to leo1)
    MAC address:  00:aa:bb:cc:a9:1f
    IPv4 address: 192.168.66.44
    IPv6 address: e322:f4fc:6bdd:5ab0:7f18:3c2c:5138:3d66
    FQDN:         leo2.example.com
```
Both interfaces are managed. Render kickstart default template:
```
network --device=00:aa:bb:cc:a9:1f --hostname floyd-obanion.example.com --vlanid=tag_coconut --interfacename=vlantag_coconut --nodnsnetwork --device=00:aa:bb:cc:72:b6 --hostname floyd-obanion.example.com --nodns
```

There is only one `network` command. The second interface is not set. Also, the `ksvalidator --version RHEL9 ks.ks` throws an error:
```
unrecognized arguments: --nodnsnetwork
```

Reply:
Fixed, was missing `\n` at the end of the `network` command output. This also explains why you got only a single line: the two interfaces were fused into one line: `-nodns` `network`